dual_net.py

```diff
@@ -0,0 +1,155 @@
+import tensorflow as tf
+import tensorflow.contrib.layers as layers
+
+
+# A convolutional block as described in AlphaGo Zero
+def conv_block(tensor, specs):
+    tensor = layers.convolution2d(tensor,
+                                  num_outputs=specs['num_outputs'],
+                                  kernel_size=specs['kernel_size'],
+                                  stride=specs['stride'],
+                                  activation_fn=None)
+    tensor = tf.layers.batch_normalization(tensor)
+    tensor = tf.nn.relu(tensor)
+    return tensor
+
+
+# A residual block as described in AlphaGo Zero
+def residual_block(tensor, specs):
+    input_tensor = tensor
+    tensor = conv_block(tensor, specs)
+    tensor = layers.convolution2d(tensor,
+                                  num_outputs=specs['num_outputs'],
+                                  kernel_size=specs['kernel_size'],
+                                  stride=specs['stride'],
+                                  activation_fn=None)
+    tensor = tf.layers.batch_normalization(tensor)
+    tensor += input_tensor
+    tensor = tf.nn.relu(tensor)
+    return tensor
+
+
+def build_mlp(board_placeholder,
```

Comment:
Just curious, why did you call this `build_mlp`? I thought "mlp" referred to specifically a NN made up of fully connected layers.

Reply:
No reason other than that's what it was called in the homeworks lol

Reply:
I changed it to "build_model" which I think is more honest 😛